guides/apm-release-process.md

```diff
@@ -0,0 +1,164 @@
+    possible to publish a new version with the same version number.
+    * if a commit is extremely simple (remove console.log, change comments) i often commit
+    directly to master.
+    * if a commit only impacts testing or files not included in a published release (excluded by
+    `.npmignore`) a PR may be merged without review.
+
+### auxillary tools
+
+There are a number of packages that are used to facilitate testing the node agent for release.
+Those that are used in the released product are enumerated in package.json's dependencies section.
+But not all the packages used for more exhaustive, release-focused testing are part of the
+devDependencies section.
+
+These are:
+
+- the todo application. this was originally google's todomvc-mongodb and was created as a simple
+showcase for angular. because the changes to make it a facility to test our agent were so extensive
+and of no general use the repository was cloned, and renamed, and is now `bmacnaughton/todo`.
+- `multiload`. this was originally developed, in very skeletal form, as part the coding challenge of
+bmacnaughton's interview . the can be used to drive a mixed load of transactions against the todo
+server. it's extensible but not particularly well documented. it can be found at
+`bmacnaughton/multiload`.
+- `testeachversion`. this is derived from Stephen Belanger's `alltheversions`. it is primarily used
+for the test matrix (`appoptics/node-apm-matrix`) but can be used locally for development as well (it's
+particularly useful to make sure all versions work when modifying a single package's probe file).
+`testeachversion` is driven by `test/versions.js` which defines the packages and versions of each
+package to be tested. there are a number of command line options for selecting specific packages, changing
+the location of the versions file, etc. but the primary documentation is the code. it produces two
+files - a summary file and a details file. both file names include the version of node, the os, and a
+timestamp. the details file contains the raw test output is helpful when investigating why tests failed;
+the summary file is a JSON file that can be interpreted by `humanize` (see next bullet). `testeachversion`
+can be found in `bmacnaughton/testeachversion`. (this package is part of devDependencies.)
+- `humanize-logs` is part of `testeachversion` and is mapped to `humanize` in the `node_modules/.bin/`
+directory. it interprets the content of the summary files produced by `testeachversion` either singly
+or in aggregate. various options exist and are documented in the help and, as usual, in the code.
+- `github.com/appoptics/apm-benchmark-node` is a set of files to create docker containers that run
+benchmarks of one version against another version to check for memory leaks, primarily, though
+CPU usage can also be evaluated.
+- `github.com/appoptics/apm-test-node` is a set of files for creating the matrix tests in docker containers.
```

Comment:
Similar to the `todo` project, can these tools be forked or transferred to appoptics GH org?